main/provider/index.ts

```diff
@@ -71,6 +72,12 @@ const storeApi = {
 
 const getPayloadOrigin = ({ _origin }: RPCRequestPayload) => storeApi.getOrigin(_origin)
 
+//TODO: should we handle non-EIP2612 e.g. DAI on mainnet?
+const signatureRequestRecognition = (typedMessage: TypedMessage<SignTypedDataVersion>) => {
```

Comment:
this can probably be named a little more clearly and also moved somewhere, maybe a module that deals with typed messages?